File: app/src/main/java/com/loskon/noteminimalism3/database/NoteCursorWrapper.java
package com.loskon.noteminimalism3.database;

import android.database.Cursor;
import android.database.CursorWrapper;

import com.loskon.noteminimalism3.model.Note;

import java.util.Date;

import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_DATE;
import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_DATE_DEL;
import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_DATE_MOD;
import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_FAVORITES;
import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_ID;
import static com.loskon.noteminimalism3.database.NoteDbSchema.COLUMN_TITLE;

/**
 * Класс обертки для курсора, который передает все вызовы к реальному объекту курсора
 * Основное использование для данного класса заключается в расширении курсора, при этом
 * переопределяя только подмножество его методов
 */

public class NoteCursorWrapper extends CursorWrapper {
    // Создает обертку для курсора
    public NoteCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    // Получает курсор, обернутый этим экземпляром для извлечения данных столбцов
    public Note getNotes() {

        long  id = getLong(getColumnIndex(COLUMN_ID));
        String text = getString(getColumnIndex(COLUMN_TITLE));
        long date = getLong(getColumnIndex(COLUMN_DATE));
        long dateMod = getLong(getColumnIndex(COLUMN_DATE_MOD));
        long dateDelete = getLong(getColumnIndex(COLUMN_DATE_DEL));
        String fav = getString(getColumnIndex(COLUMN_FAVORITES));

        // ВНИМАНИЕ!!!! SQLite не имеет отдельного класса логического хранилища.
        boolean favoritesItem = fav.equals("1");

        return new Note(id, text, new Date(date),
                new Date(dateMod), new Date(dateDelete), favoritesItem);
    }
}

File: app/src/main/java/com/loskon/noteminimalism3/sqlite/NoteCursorWrapper.kt
package com.loskon.noteminimalism3.sqlite

import android.database.Cursor
import android.database.CursorWrapper
import com.loskon.noteminimalism3.app.base.datetime.toLocalDateTime
import com.loskon.noteminimalism3.model.Note
import com.loskon.noteminimalism3.sqlite.NoteDatabaseSchema.NoteTable

/**
 * Извлечение данных столбцов
 */

class NoteCursorWrapper(cursor: Cursor) : CursorWrapper(cursor) {

    fun getNotes(): Note {
        val id = getLong(getColumnIndex(NoteTable.COLUMN_ID))
        val title = getString(getColumnIndex(NoteTable.COLUMN_TITLE))
        val dateCreation = getLong(getColumnIndex(NoteTable.COLUMN_DATE))
        val dateModification = getLong(getColumnIndex(NoteTable.COLUMN_DATE_MOD))
        val dateDelete = getLong(getColumnIndex(NoteTable.COLUMN_DATE_DEL))
        val isFavorite = getInt(getColumnIndex(NoteTable.COLUMN_FAVORITES)) == 1
        val isDelete = getInt(getColumnIndex(NoteTable.COLUMN_DEL_ITEMS)) == 1

        return Note(
            id = id,
            title = title,
            createdDate = dateCreation.toLocalDateTime(),
            modifiedDate = dateModification.toLocalDateTime(),
            deletedDate = dateDelete.toLocalDateTime(),
            isFavorite = isFavorite,
            isDeleted = isDelete
        )
    }
}